src/github3/checks.py

```diff
@@ -103,13 +103,24 @@ class CheckApp(models.GitHubCore):
         representing the app owner
     """
 
+    CUSTOM_HEADERS = {
+        "Accept": "application/vnd.github.machine-man-preview+json"
+    }
+
     def _update_attributes(self, app):
         self.description = app["description"]
         self.external_url = app["external_url"]
         self.html_url = app["html_url"]
         self.id = app["id"]
         self.name = app["name"]
         self.owner = app["owner"]
+        _, self.slug = app["html_url"].rsplit("/", 1)
```

Comment:
I'm sorry if I wasn't clear. I don't like generating false attributes like this that aren't actually present in the response. We also generally avoid properties except in a few internal cases, which this is not.

Instead let's do

```suggestion
        _, slug = self.html_url.rsplit("/", 1)
        self._api =self._build_url("apps", slug)
```

And in the `to_app` method we can do `self._get(self._api, headers=headers)`

Reply:
I was trying to avoid calling `self._build_url(...)` here since this opens another can of worms due to `self._build_url()` being only mocked in the class under test but not on the mock session that is passed around. I'll proxy the `build_url()` of the session now to the real implementation.

Reply:
Ok, so this wasn't as bad as it looked on the first sight :wink: